File: solutions/problem_7/solution.py
import sys
import os
sys.path.append(os.path.abspath(os.path.dirname('./utils/python')))
from python.utils import timeit, is_prime

LIMIT = 10001

@timeit
def solution():
    number = 1
    counter = 2  # two is prime
    while (counter <= LIMIT):
        number += 2
        if is_prime(number):
            counter += 1
    return number


solution()

File: solutions/problem_7/solution.c
#include <stdio.h>
#include "utils.h"

#define LIMIT 10001

int solution(){
    int number, counter;

    number = 1;
    counter = 2;

    while(counter <= LIMIT){
        number = number + 2;
        if(is_prime(number) == 1){
            counter = counter + 1;
        }
    }

    return number;
}

int main(int argc, char *argv[]){
    return int_time_it(solution);
}